nx-X11/programs/Xserver/hw/nxagent/Events.c

```diff
@@ -2502,7 +2502,7 @@ FIXME: This can be maybe optimized by consuming the
                        nxagentExposeQueue.exposures[index].remoteRegion, &sum);
 
       #ifdef TEST
-      fprintf(stderr, "nxagentHandleExposeEvent: Added region for window [%ld] to position [%d].\n",
```

Comment:
Please drop this whole commit.

I know we've been "fixing" these issues before, but that probably was a very bad idea to begin with. We're just shadowing *real* problems with that.

`nxagentWindow()` is a macro for getting a `Window` object, which can either be `CARD32` in X server or `unsigned long` (which does have a variable, machine-dependent size) in Xlib context. Provided that we pass that to an Xlib function later on, the code is wrong (or rather, the original code uses the "correct" size specifier in the `fprintf()` call, but falls short of handling everything else correctly...)

Let's not mask this.

Reply:
So what's your suggestion to get rid of the compilation errors in that case?

Reply:
Hmm, let me think about that and look through more code (including pre-processed one) and **do not** drop this commit.

It might be fine after all.

More comments later, I guess.

Reply:
Your modified hunk can stay.

`nxagentWindow(pWin)` is not used in other places than this message.